### ISSUE TITLE
Run single rule *without* using `--no-eslintrc`

### ISSUE THREAD

**Issue by phenomnomnominal** [eslint/eslint#12666]
**The version of ESLint you are using.**

6.5.1

**The problem you want to solve.**

I would like to be able to run a single ESLint rule, but without having to repeat all the other setup config that is in my root `.eslintrc.js` file.

My specific use case is running a single file on a TypeScript file, something like this:

```ts
console.log('foo');
debugger;
export enum Numbers {
  one,
  two,
  three,
  four
}

```

Running `eslint --no-eslintrc --rule no-debugger:error` fails on this file with a parsing error (the enum).

Running `eslint --rule no-debugger:error` runs all the other rules defined in the `extends` part of my root `.eslintrc.js` file.

This problem is not isolated to the CLI, but also the JS API, which I am currently using.

**Your take on the correct solution to problem.**

I believe I could use the JS API to do `resolveFileGlobPatterns()` and use `getConfigForFile()` for each file and then modify the result to just set the one rule I want, but that seems like it would be really slow. Is that the correct approach? Is there something I am missing?

**Are you willing to submit a pull request to implement this change?**

Yes, happy to give it a crack!

**Comment by ilyavolodin** [comment]
This should be easily solvable with splitting your config into two pieces. One that describes general setup, and another with all rule configuration. You can then extend rules config with the general config. When you want to run single rule, you can use `--no-eslintrc` command and supply a direct reference to general config using `--config` flag. I know this is going to be a long command line, but honestly, this seems like a supper edge-casy scenario to build something like this into ESLint.

**Comment by phenomnomnominal** [comment]
So for even more context, I first came across this issue while trying to run a single rule on a really large project in a monorepo, with multiple ESLint configs, where it wouldn't be that simple to rearrange the configs. The goal was to incrementally introduce one new rule on the codebase with a separate command.

Is it really that edge-casey to want to run a single rule on a single file with all other config in tact?

**Comment by ljharb** [comment]
This would be immensely valuable; using `--no-eslintrc` bypasses custom parser settings, custom plugin settings, envs, globals, etc.

In other words, what's needed imo is a `--disable-all-rules` flag that can be combined with `--rule`.

@ilyavolodin I use this *constantly* to try to autofix or test a single rule in anything using my shared configs, all of which abstract away a ton of custom settings that are a huge PITA to configure via the CLI.

**Comment by kaicataldo** [comment]
This seems useful to me for the case brought up in https://github.com/eslint/eslint/issues/12666#issuecomment-565829367. I've generally been the point-person on the teams I've been when we would introduce new ESLint rules, so I've felt this pain.

**Comment by kaicataldo** [comment]
Also, since this is a change to core, I think this proposal should go through the RFC process outlined [here](https://github.com/eslint/rfcs).

**Comment by ilyavolodin** [comment]
OK, so maybe it's not as edge-casy as I though it is. In that case, as @kaicataldo mentioned, this would need to go through RFC process. I just really hate adding more and more flags to CLI, but if they are justified - I guess that changes things.
